src/dialects/firebird/transaction.js

```diff
@@ -0,0 +1,41 @@
+
+var Transaction = require('../../transaction')
+var assign      = require('lodash/object/assign');
+var inherits    = require('inherits')
+var debug       = require('debug')('knex:tx')
+var helpers     = require('../../helpers')
+
+function Transaction_Firebird() {
+  Transaction.apply(this, arguments)
+}
+inherits(Transaction_Firebird, Transaction)
+
+assign(Transaction_Firebird.prototype, {
+
+  query: function(conn, sql, status, value) {
+    var t = this
+    var q = this.trxClient.query(conn, sql)
+      .catch(function(err) {
+        return err.errno === 1305
+      }, function() {
+        helpers.warn('Transaction was implicitly committed, do not mix transactions and DDL with Firebird (#805)')
```

Comment:
I don't really understand the conditions under which this warning might arise.

Could you please alias 1305 with a descriptive constant name? (It can be in this file)

Also I'm not sure which part of #805 is relevant. Could you include detail that is meant to be communicated via 805 directly into this warning message pls?